src/main/java/daomephsta/unpick/impl/IntegerType.java

```diff
@@ -1,11 +1,59 @@
 package daomephsta.unpick.impl;
 
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
 import org.objectweb.asm.*;
 import org.objectweb.asm.tree.AbstractInsnNode;
 import org.objectweb.asm.tree.InsnNode;
 
 public enum IntegerType
 {
+	BYTE(Byte.class, byte.class, Type.BYTE_TYPE, Opcodes.IAND, Opcodes.IRETURN) 
+	{
+		@Override
+		public AbstractInsnNode createLiteralPushInsn(long literal)
+			{ return InstructionFactory.pushesInt((byte) literal); }
+
+		@Override
+		public void appendLiteralPushInsn(MethodVisitor mv, long literal)
+			{ InstructionFactory.pushesInt(mv, (byte) literal); }
+		
+		@Override
+		public Number box(long value)
+			{ return Byte.valueOf((byte) value); }
+
+		@Override
+		public Number binaryNegate(Number value)
+			{ return ~value.byteValue(); }
```

Comment:
This will return an Integer because the `~` operator implicitly widens to int. So it needs casting back to `byte` after the operation.